### PR TITLE
Add Delete function to persistent storage

### PR DESCRIPTION
**Pull request by nkuba**
Expose a Delete function that can be used to remove file from persistent sotarge. Add disk storage implementation.

### CODE REVIEW

File: pkg/persistence/disk_persistence.go [hunk 1]
```diff
@@ -149,6 +149,13 @@ func (ds *diskPersistence) Archive(directory string) error {
 	return moveAll(from, to)
 }
 
+func (ds *diskPersistence) Delete(dirName string, fileName string) error {
+	dirPath := ds.getStorageCurrentDirPath()
+	filePath := filepath.Join(dirPath, dirName, fileName)
```
Comment (pdyraga) [inline pkg/persistence/disk_persistence.go:154]: Can we take the opportunity and update all `fmt.Sprintf("%s/%s/%s",...)` occurrences in this file?

File: pkg/persistence/disk_persistence_test.go [hunk 2]
```diff
@@ -499,3 +499,55 @@ func TestDiskPersistence_AppendToArchive(t *testing.T) {
...
+
+	bytesToTest := []byte{115, 111, 109, 101, 10}
+
+	diskPersistence.Save(bytesToTest, dirName1, fileName11)
+
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		if err != nil {
+			t.Fatalf("Dir [%+v] was supposed to be created", path)
+		}
+	}
```
Comment (pdyraga) [inline pkg/persistence/disk_persistence_test.go:515]: Shouldn't we instead check if the file was created?

Something like:
```
	pathToFile := fmt.Sprintf("%s/%s/%s", pathToCurrent, dirName1, fileName11)

	if _, err := os.Stat(pathToFile); os.IsNotExist(err) {
		t.Fatalf("file [%+v] was supposed to be created", pathToFile)
	}
```

File: pkg/persistence/disk_persistence_test.go [hunk 3]
```diff
@@ -499,3 +499,55 @@ func TestDiskPersistence_AppendToArchive(t *testing.T) {
...
+	err := diskPersistence.Delete(dirName1, fileName11)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	if _, err := os.Stat(path); !os.IsNotExist(err) {
+		if err != nil {
+			t.Fatalf("Dir [%+v] was supposed to be moved", path)
+		}
+	}
```
Comment (pdyraga) [inline pkg/persistence/disk_persistence_test.go:526]: I am confused. I would expect the file to be deleted, not the directory. See that `path` points to the directory not to the file. Also s/moved/deleted.

File: pkg/persistence/disk_persistence.go (hunk 1 above)
Comment (nkuba) [reply pkg/persistence/disk_persistence.go:154]: 41ca4fdc902a4c0b4c13d0007d811859bf9c9ba0

File: pkg/persistence/disk_persistence_test.go (hunk 2 above)
Comment (nkuba) [reply pkg/persistence/disk_persistence_test.go]: The tests were confusing, I updated them in 286069a18defe0155fc24fd7b047eff5cafe4a2f and 22a0f19484663ed9a39425df79aa9c09ad154798.

File: pkg/persistence/disk_persistence_test.go (hunk 3 above)
Comment (nkuba) [reply pkg/persistence/disk_persistence_test.go]: 💯 see https://github.com/keep-network/keep-common/pull/103#discussion_r969445720

**Review by pdyraga — APPROVED** [review]

**Pull request merged**